.github/workflows/publish-python.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish Python Package
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  pypi-publish:
+    name: Publish release to PyPI
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for sigstore
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.9"
+      - name: Install dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          pip3 install wheel
+      - name: Build package
+        run: |
+          python3 setup.py sdist bdist_wheel
```

Comment:
I am afraid we are somewhat inconsistent how we build the package. Specifically, we use `setup.py` here but `pip wheel` in the Docker container. For other projects, we typically use [build](https://pypi.org/project/build/). Unless there is a specific reason to keep these as-is, I would prefer we use `build` both in the *Publish to PyPI* and the Docker distribution actions.